lib/src/revset.rs

```diff
@@ -98,6 +97,15 @@ pub enum RevsetEvaluationError {
     Other(Box<dyn std::error::Error + Send + Sync>),
 }
 
+impl RevsetEvaluationError {
+    pub fn into_backend_error(self) -> BackendError {
+        match self {
+            Self::StoreError(err) => err,
+            Self::Other(err) => BackendError::Other(err),
```

Comment:
It's probably better to do `panic!()` as before because that's the caller's expectation. The function can be renamed to `.expect_backend_error()`?

Reply:
This somewhat defeats the point of the PR; the goal of being able to return errors from the Revset iterator is as a safer and cleaner alternative to panics.  If the client panics anyway this achieves a lot less.

I'll leave it like this for now since I'm starting to accrue conflicts but I'd like to address this in a future PR.

Reply:
If revset engine can return `Other(err)` for these ancestry queries, callers would have to propagate it as `RevsetEvaluationError`. That's the point ,and leaving TODO looks good to me. Thanks.